sql/analyzer/indexed_joins.go

```diff
@@ -320,38 +320,58 @@ func addRightSemiJoins(m *Memo) error {
 // candidate indexes as replacements for the given relExpr, or empty values
 // if there are no suitable indexes.
 func lookupCandidates(ctx *sql.Context, rel relExpr, aliases TableAliases) (string, []sql.Index, error) {
-	var attributeSource string
-	var indexableTable sql.IndexAddressableTable
-	var ok bool
 	switch n := rel.(type) {
 	case *tableAlias:
-		attributeSource = strings.ToLower(n.table.Name())
-		rt, ok := n.table.Child.(*plan.ResolvedTable)
-		if !ok {
-			return "", nil, nil
-		}
-		table := rt.Table
-		if w, ok := table.(sql.TableWrapper); ok {
-			table = w.Underlying()
-		}
-		indexableTable, ok = table.(sql.IndexAddressableTable)
-		if !ok {
-			return "", nil, nil
-		}
-		aliases.add(n.table, indexableTable)
+		return tableAliasLookupCand(ctx, n.table, aliases)
 	case *tableScan:
-		attributeSource = strings.ToLower(n.table.Name())
-		table := n.table.Table
-		if w, ok := table.(sql.TableWrapper); ok {
-			table = w.Underlying()
-		}
-		indexableTable, ok = table.(sql.IndexAddressableTable)
-		if !ok {
-			return "", nil, nil
+		return tableScanLookupCand(ctx, n.table, aliases)
+	case *selectSingleRel:
+		switch t := n.table.Rel.(type) {
+		case *plan.TableAlias:
+			return tableAliasLookupCand(ctx, t, aliases)
+		case *plan.ResolvedTable:
+			return tableScanLookupCand(ctx, t, aliases)
+		default:
 		}
 	default:
+	}
+	return "", nil, nil
+
+}
+
+func tableScanLookupCand(ctx *sql.Context, n *plan.ResolvedTable, aliases TableAliases) (string, []sql.Index, error) {
+	attributeSource := strings.ToLower(n.Name())
+	table := n.Table
+	if w, ok := table.(sql.TableWrapper); ok {
+		table = w.Underlying()
+	}
+	indexableTable, ok := table.(sql.IndexAddressableTable)
+	if !ok {
+		return "", nil, nil
+	}
+	aliases.add(n, indexableTable)
```

Comment:
Just making certain we meant to add this here; in the old *tableScan case I don't think we did...